history/generate_plot.py

```diff
@@ -0,0 +1,29 @@
+import locale
+
+import matplotlib.pyplot as plt
+import matplotlib.ticker as mtick
+import pandas as pd
+
+from constants import COLUMN_START_DATE, COLUMN_GASOLINA_IO95, COLUMN_GASOLINA_IO98, COLUMN_GASOLEO_RODOVIARIO, HISTORY_PLOT_LABEL_GASOLINA_IO95, HISTORY_PLOT_LABEL_GASOLINA_IO98, HISTORY_PLOT_LABEL_GASOLEO_RODOVIARIO, HISTORY_PLOT_Y_LABEL, HISTORY_PLOT_X_LABEL
+
+locale.setlocale(locale.LC_ALL, 'pt_PT.UTF-8')
+
+
+def generate_plot_history(history_file, plot_location):
+    # Read history
+    history = pd.read_csv(history_file)
+
+    # Convert start dates and end dates to datatime
+    history[COLUMN_START_DATE] = pd.to_datetime(history[COLUMN_START_DATE])
+
+    # Select the last 6 months of data
+    history.set_index(COLUMN_START_DATE, inplace=True)
+    history = history.last('6M').copy()
+
+    # Plot
+    plt.figure()
+    ax = history.plot(y=[COLUMN_GASOLINA_IO95, COLUMN_GASOLINA_IO98, COLUMN_GASOLEO_RODOVIARIO],
+                      label=[HISTORY_PLOT_LABEL_GASOLINA_IO95, HISTORY_PLOT_LABEL_GASOLINA_IO98, HISTORY_PLOT_LABEL_GASOLEO_RODOVIARIO], ylabel=HISTORY_PLOT_Y_LABEL, xlabel=HISTORY_PLOT_X_LABEL)
+
+    ax.yaxis.set_major_formatter(mtick.FormatStrFormatter('%.3f'))
+    plt.savefig(plot_location, dpi=300, bbox_inches='tight')
```

Comment:
Do we really need to save the plots as a picture in out git repository, or can we just have an image object that we can upload to Twitter straight in the Github Action? For me it's perfectly fine like this. 

Also since, it's expected to the image to be there, if the image does not exist it may lead to problems.

What I purpose as alterantive is saving it as an image object, and post the buffer directly.

_Here's an example:_

```python
from io import BytesIO
buffer = BytesIO()

plt.savefig(buffer,format='png')
PNG = buffer.getvalue()
```

What do you think?

Reply:
I think what @joaoofreitas is saying makes some sense. We probably only need to generate the image to upload it to Twitter, and then we can delete it. 

We're still keeping the gas data to create our desired plot images. 

Let us know your thoughts, @Dntfreitas.

Reply:
Well... it doesn't hurt to have the image saved every week. 

Also, since we are going to publish the plot monthly, if you want to include this image in another place (say, e.g., an external website), you should  get the updated image every week.

Reply:
__IMHO__ if we want to access the data for any external source, we have all the data in the repository, if we want the images we can just use Twitter _API_  to retrieve them. If we want to store the images in the repository either we store all of them or none. 

The implementation is based in just a couple of lines, that store the image in a buffer in RAM instead of the filesystem.

It's just my opinion, for me it's approved anyways. The main focus is the functionality and your work has been exceptional. @Dntfreitas 

Any thoughts? @carlosrsabreu @13dev

Reply:
agree, the repo already store the graph data with this info you can recreate how many images you want (1y, 3 weeks, etc) its useless to store images. just create the image on the fly

with that you can create a bot command to retrive such info, ex metion on twitter "@devo-abastecer info 13-04 13-09" - retrive all data since 13-04 to 13-09, and so on